frontend-angularjs/app/scripts/controllers/stationsCtrl.js

```diff
@@ -142,6 +113,70 @@ angular.module('hermesApp').controller('StationsCtrl', function ($scope, $http,
   };
 });
 
+angular.module('hermesApp').controller('StationLocPopoverCtrl', function ($scope, $timeout) {
+  const GMAP_ID_PREFIX = "gmap-loc";
+  const GMAPS_GEOCODER = new google.maps.Geocoder();
+
+  $scope.location = {};
+
+  $scope.initLocPopover = function (index, label, lat, lng) {
+    $scope.location.label = label;
+    $scope.location.lat = lat;
+    $scope.location.lng = lng;
+    $scope.location.gmapId = GMAP_ID_PREFIX + index;
+
+    decodeAddress(lat, lng, function (address) {
+      $timeout(function () {
+        $scope.location.address = address;
+      });
+    });
+  };
+
+  $scope.displayLocation = function (label, lat, lng) {
+    $timeout(function () {
+      const GMAP_EL = document.getElementById($scope.location.gmapId);
+
+      const locGMapStationLatLng = new google.maps.LatLng(lat, lng);
+      const locGMapObj = new google.maps.Map(GMAP_EL, {
+        zoom: 18,
+        center: locGMapStationLatLng,
+        scrollwheel: false
+      });
+      const locGMapStationTooltip = new google.maps.InfoWindow({
+        content: label
+      });
+      const locGMapStationMarker = new google.maps.Marker({
+        position: locGMapStationLatLng,
+        map: locGMapObj
+      });
+
+      locGMapStationTooltip.open(locGMapObj, locGMapStationMarker);
+    });
+  };
+
+  function decodeAddress(lat, lng, callback) {
+    if (!lat || !lng)
+      callback("N/A");
+
+    var latlng = new google.maps.LatLng(lat, lng);
+    GMAPS_GEOCODER.geocode({
+      'latLng': latlng
+    }, function (results, status) {
+      if (status === google.maps.GeocoderStatus.OK) {
+        if (results[1]) {
```

Comment:
Does it makes sense to put this condition in previous line to avoid two similar 'else' blocks?